Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1

```diff
@@ -344,72 +386,100 @@ function Remove-AzTSInvalidAADAccounts
+        $invalidClassicRoles | ForEach-Object {
+            try 
             {
-                $armUri = "https://management.azure.com" + $_.RoleAssignmentId + "?api-version=2015-06-01"
-                $method = "Delete"
-                $classicAssignments = $null
-                $classicAssignments = [ClassicRoleAssignments]::new()
-                $headers = $classicAssignments.GetAuthHeader()
-                $res = $classicAssignments.DeleteClassicRoleAssignmnets([string] $armUri, [string] $method,[psobject] $headers)
-
-                if(($null -ne $res) -and ($res.StatusCode -eq 202 -or $res.StatusCode -eq 200))
+                if($_.RoleDefinitionName -eq "CoAdministrator" -and $_.RoleAssignmentId.contains("/providers/Microsoft.Authorization/classicAdministrators/"))
                 {
-                    $_ | Select-Object -Property "SignInName", "RoleAssignmentId", "RoleDefinitionName"
-                }
-            } 
+                    $armUri = "https://management.azure.com" + $_.RoleAssignmentId + "?api-version=2015-06-01"
+                    $method = "Delete"
+                    $classicAssignments = $null
+                    $classicAssignments = [ClassicRoleAssignments]::new()
+                    $headers = $classicAssignments.GetAuthHeader()
+                    $res = $classicAssignments.DeleteClassicRoleAssignmnets([string] $armUri, [string] $method,[psobject] $headers)
+
+                    if(($null -ne $res) -and ($res.StatusCode -eq 202 -or $res.StatusCode -eq 200))
+                    {
+                        $_ | Select-Object -Property "SignInName", "RoleAssignmentId", "RoleDefinitionName"
+                    }
+                } 
+            }
+            catch
+            {
+                $isRemoved = $false
+                Write-Host "Not able to remove invalid classic role assignment. ErrorMessage [$($_)]" -ForegroundColor Red  
+            }
         }
-        catch
+
+        if($isRemoved)
         {
-            $isRemoved = $false
-            Write-Host "Not able to remove invalid classic role assignment. ErrorMessage [$($_)]" -ForegroundColor Red  
+            Write-Host "Completed deleting invalid AAD Object guids role assignments." -ForegroundColor Green
+        }
+        else 
+        {
+            Write-Host "`n"
+            Write-Host "Not able to successfully delete invalid AAD Object guids role assignments." -ForegroundColor Red
         }
     }
-
-    if($isRemoved)
-    {
-        Write-Host "Completed deleting invalid AAD Object guids role assignments." -ForegroundColor Green
-    }
-    else 
+    else
     {
-        Write-Host "`n"
-        Write-Host "Not able to successfully delete invalid AAD Object guids role assignments." -ForegroundColor Red
-    }
+        Write-Host "Validate all role assignments that are going to remediate using this script at [$($folderPath)]..." -ForegroundColor Green
```

Comment:
Output from a `DryRun` can become outdated by the time the actual remediation takes place. This is because, the assignments are collected afresh every time the script is run, and users can see assignments not present in the output of `DryRun` getting remediated as well - Though this is valid, users might wrongly interpret this as an inconsistency.

Do you want to remove only those assignments that were generated by the `DryRun` command?

Reply:
Added -FilePath parameter to remediate role assignments generated by -DryRun output.